binaries/scoot-snapshot-db/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/scootdev/scoot/os/temp"
+	"github.com/scootdev/scoot/snapshot"
+	"github.com/scootdev/scoot/snapshot/cli"
+	"github.com/scootdev/scoot/snapshot/git/gitdb"
+	"github.com/scootdev/scoot/snapshot/git/repo"
+)
+
+func main() {
+	inj := &injector{}
+	cmd := cli.MakeDBCLI(inj)
+	if err := cmd.Execute(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type injector struct{}
+
+func (i *injector) RegisterFlags(rootCmd *cobra.Command) {}
+
+func (i *injector) Inject() (snapshot.DB, error) {
```

Comment:
So, I maybe buried the lede:

In all our other CLI tools, we've had a problem trying to mix args. So we've had to setup first, then call the CLI, and trying to combine has gotten messy (especially with the --addr flags).

This is nifty because it allows us to do everything in cobra: the injector can register flags (though doesn't right now), and then those get parsed naturally by cobra.

So, at some point (and soon), we're going to want the injector to have args, and it's going to be able to; yay!

To summarize:
the injector doesn't just delay db construction; it allows getting args from the CLI, using them to make objects, and *then* calling into the CLI's shared logic, across our open-source and closed-source binaries.